### DISCUSSION TITLE
Moving preexisting SDL-based formats to KDL

### DISCUSSION THREAD

**Discussion by ZILtoid1991** [kdl-org/kdl#394] [General]
My game engine (PixelPerfectEngine) uses multiple SDL-based formats for configuration, map data, etc., and I find the SDL-based approach a bit lacking at certain things. Originally I wanted to write my own solution, but after seeing this format, I'm thinking on moving to this one instead. My only gripes so far are the lack of mention of the namespaces (used for nodes, not used for attributes), and no anonymous nodes (used in certain situations as mass-data storage), I don't know if I should break some rules with my own implementation for D (which is likely the current biggest user of SDL for its package manager dub), or do some adjustments to my own formats besides getting rid of the occasional `L` at the end of certain values.

**Comment by tabatkins** [comment]
Could you explain more about the things you're running into trouble with?

* For namespaces, how much is solved by just uniquely naming nodes, like `foo:bar`? This doesn't let you query for "all bar nodes" or "all nodes in the foo namespace", but is that something you need to do?
* Alternately, for namespaces, how much is solved by putting a namespace in the node tag, like `(foo)bar`? Are you planning to use node tags for anything else?
* Can you elaborate on how you are using anonymous nodes? The usual solution is to just use a placeholder name, like `*` or `-` for the nodes.

(Also, by "SDL-based", do you mean SDLang? There's a bunch of formats using the acronym "SDL", so I'm just making sure.)

**Reply by zkat** [reply]
also worth noting: `""` is a perfectly valid node name.
